source/Octopus.Tentacle/Commands/RunAgentCommand.cs

```diff
@@ -164,19 +156,9 @@ protected override void Stop()
                 halibut.Value.Stop();
             }
 
-            if (workspaceCleanerHasStarted)
-            {
-                workspaceCleanerTask.Value.Stop();
-            }
-
-            if (kubernetesPodMonitorTask.IsValueCreated)
-            {
-                kubernetesPodMonitorTask.Value.Stop();
-            }
-
-            if (kubernetesOrphanedPodCleanerTask.IsValueCreated)
+            foreach (var backgroundTaskLazy in backgroundTasks.Where(bt => bt.IsValueCreated))
```

Comment:
We can just check to see if the Lazy value has been created before we call `Stop()`.